server/functions/binary/equal.go

```diff
@@ -468,9 +468,15 @@ var numeric_eq = framework.Function2{
 }
 
 // oideq_callable is the callable logic for the oideq function.
+// This method doesn't use DotlgresType.Compare because it's on the critical path for many tooling queries that
+// examine the pg_catalog tables.
 func oideq_callable(ctx *sql.Context, _ [3]*pgtypes.DoltgresType, val1 any, val2 any) (any, error) {
-	res, err := pgtypes.Oid.Compare(ctx, val1.(id.Id), val2.(id.Id))
```

Comment:
Can the `Compare` method apply a similar change to directly compare the `id.Id`s instead of using `id.Cache().ToOID()`, too?

Reply:
I don't think so, because Compare is generalized for sorting and turns strings into integers. Equality is a special case where it doesn't matter and you can do something more efficient.